mod-infra/src/main/scala/at/forsyte/apalache/infra/passes/options.scala

```diff
@@ -2,10 +2,25 @@ package at.forsyte.apalache.infra.passes.options
 
 import java.io.File
 import at.forsyte.apalache.tla.lir.Feature
+import com.google.inject
+import scala.util.{Failure, Success, Try}
+import at.forsyte.apalache.infra.PassOptionException
 
 /**
  * The components of this package specify the configurations and options used to configure Apalache
  *
+ * Configurations are represented by extension of the [[Config]] trait. Configurations are derived via injective maps
+ * from configuration sources (souch as CLI arguments or .cfg files) to config classes. As such, each value in a config
+ * class is an `Option`, where a value of `None` indicates that value was left unconfigured.
+ *
+ * Options are represeneted by extensions of the [[OptionGroup]] trait. Option groups are typically derived via
+ * surjective maps from from [[Config]]s. As such, for every field in the option group, there must be a value given in
```

Comment:
The OptionGroup is the co domain. That’s meant to be indicated by the “derived via”: we map out of the Config into an OptionGroup.

Is there a concise rephrasing that you think would make this clearer?

Reply:
Maybe I was just confused because in the similar paragraph about `Config` above you explicitly stated the codomain (so I assumed it's not implicitly obvious).

Reply:
Ah yeah, that makes sense. I made the parallel more consistent. Thanks.